library/src/cache/patch_manager.rs

```diff
@@ -367,28 +380,14 @@ impl ManagePatches for PatchManager {
         self.save_patches_state()
     }
 
-    fn record_boot_success_for_patch(&mut self, patch_number: usize) -> Result<()> {
-        let next_boot_patch = self
+    fn record_boot_success(&mut self) -> Result<()> {
+        let boot_patch = self
             .patches_state
-            .next_boot_patch
-            .context("No next_boot_patch")?;
-
-        if next_boot_patch.number != patch_number {
-            bail!(
-                "Attempted to record boot success for patch {} but next_boot_patch is {}",
-                patch_number,
-                next_boot_patch.number
-            );
-        }
-
-        if let Some(current_patch) = self.patches_state.last_booted_patch {
-            if current_patch.number != patch_number {
-                // If we now have a new last_booted_patch, delete the old one's artifacts.
-                let _ = self.delete_patch_artifacts(current_patch.number);
-            }
-        }
+            .last_attempted_patch
```

Comment:
Is "last_attempted_patch" the "current patch"?

Reply:
In this state, yes. This is the implementation of the `ManagePatches` trait, so all of the docs are above.